workers/unity/Packages/com.improbable.gdk.core/Worker/DefaultWorkerConnector.cs

```diff
@@ -14,58 +16,151 @@ public abstract class DefaultWorkerConnector : WorkerConnector
         ///     Determines whether to connect via the locator.
         /// </summary>
         /// <returns>True, if should connect via the Locator, false otherwise.</returns>
-        protected override bool ShouldUseLocator()
+        protected override ConnectionService GetChosenService()
         {
+            var commandLineArguments = Environment.GetCommandLineArgs();
+            var commandLineArgs = CommandLineUtility.ParseCommandLineArgs(commandLineArguments);
+
+            if (commandLineArgs.ContainsKey(RuntimeConfigNames.SteamDeploymentTag) ||
+                commandLineArgs.ContainsKey(RuntimeConfigNames.SteamTicket))
+            {
+                return ConnectionService.Locator;
+            }
+
+            if (commandLineArgs.ContainsKey(RuntimeConfigNames.LoginToken))
+            {
+                return commandLineArgs.ContainsKey(RuntimeConfigNames.PlayerIdentityToken)
+                    ? ConnectionService.AlphaLocator
+                    : ConnectionService.Locator;
+            }
+
+            return ConnectionService.Receptionist;
+        }
+
+        protected override ConnectionParameters GetConnectionParameters(string workerType, ConnectionService service)
+        {
+            // UseExternalIp needs to be true when using the locator
+            var useExternalIp = service != ConnectionService.Receptionist || UseExternalIp;
```

Comment:
Hmm.. should we print a warning if we are overriding the user chosen value?

Reply:
could do, though that warning will always trigger when launching a client via locator